docs/nima/testing.adoc

```diff
@@ -0,0 +1,397 @@
+
+.Routing test using `@RoutingTest` and `DirectClient`.
+[source, java]
+----
+@RoutingTest    <1>
+class RoutingTest {
+
+    private final Http1Client client;
+
+    protected RoutingTest(DirectClient client) { <2>
+        this.client = client;
+    }
+
+    @SetUpRoute <3>
+    static void routing(HttpRouting.Builder builder) {
+        QuickstartMain.routing(builder);
+    }
+
+    @Test <4>
+    void testRootRoute() {
+        try (Http1ClientResponse response = client.get("/greet")
+                .request()) {   <5>
+
+            JsonObject json = response.as(JsonObject.class);    <6>
+            assertThat(json.getString("message"), is("Hello World!"));
+        }
+    }
+}
+----
+<1> Use `@RoutingTest` to trigger the testing framework;
+<2> Inject `DirectClient` for the test;
+<3> SetUp routing for the test;
+<4> A regular `JUnit` test method;
+<5> Call the `client` to obtain server response;
+<6> Perform necessary assertions;
+
+If only routing tests are required, this is a "lighter" way of testing since the framework will not configure and run the full Níma server. This way, no real ports will be opened. All the communication will be done through `DirectClient` which makes the tests very effective.
+
+It is required to annotate the test class with `@RoutingTest` annotation to trigger the server to do the configuration. Thus, it will inject the `DirectClient`, which can then be used in unit tests.
+
+Routing is configured the same way as in full server testing using the `@SetUpRoute` annotation.
+
+== Additional Information
+
+=== WebSocket Testing
+
+In case WebSocket testing is required, there is an additional module for this. It is necessary to include the following Maven dependency to Project's pom file:
+
+[source,xml]
+----
+<dependency>
+    <groupId>io.helidon.nima.testing.junit5</groupId>
+    <artifactId>helidon-nima-testing-junit5-websocket</artifactId>
+    <scope>test</scope>
+</dependency>
+----
+
+
+=== WebSocket Testing Example
+
+WebSocket Testing extension adds support for routing configuration and injection of WebSocket related artifacts, such as `WebSockets` as well `DirectWsClient` in Helidon Níma unit tests.
+
+.WebSocket sample test.
+[source,java]
+----
+@ServerTest
+class WsSocketTest {
+
+    private static final ServerSideListener WS_LISTENER = new ServerSideListener();
+
+    private final WsClient wsClient;    <1>
+
+    protected WsSocketTest(WsClient wsClient) {
+        this.wsClient = wsClient;
+    }
+
+    @SetUpRoute
+    static void routing(WsRouting.Builder ws) { <2>
+        ws.endpoint("/testWs", WS_LISTENER);
+    }
+
+    @Test
+    void testWsEndpoint() { <3>
+        WS_LISTENER.reset();
+
+        ClientSideListener clientListener = new ClientSideListener();
+        wsClient.connect("/testWs", clientListener);    <4>
+
+        assertThat(clientListener.message, is("ws"));   <5>
+    }
+}
+----
+<1> Declare `WsClient` and later inject it in the constructor;
```

Comment:
These setneces should end with a period rather than a semi-colon.

<2> Using `@SetUpRoute`, create WebSocket routing and assign a serverside listener.
<3> Test the WebSocket endpoint using the regular `@Test` annotation. (added the article 'the' after using)
 <4> Create and assign the clientside listener. (added the article 'the' after assign)

Reply:
ok)